docs/real-time-analytics/discounting/process.sql

```diff
@@ -0,0 +1,64 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Create Discount Code Table
+CREATE TABLE DISCOUNT_CODES (
+    CODE VARCHAR PRIMARY KEY, 
+    PERCENTAGE DOUBLE
+) WITH (
+    KAFKA_TOPIC = 'discount_codes',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+);
+
+-- Create Order Stream
+CREATE STREAM ORDER_STREAM (
+	ID BIGINT, 
+	ORDER_TIME TIMESTAMP, 
+	CUSTOMER_ID BIGINT, 
+	ITEM_CT INT, 
+	ORDER_SUBTOTAL DOUBLE, 
+	DISCOUNT_CODE VARCHAR
+) WITH (
+  KAFKA_TOPIC = 'orders',
+  VALUE_FORMAT = 'JSON',
+  PARTITIONS = 6
+);
+
+
+-- Application logic
+
+-- Enrich the orders stream with discount details.
+-- Compute the discount amount as well as order total.
+CREATE STREAM ORDERS_ENRICHED WITH (
+	KAFKA_TOPIC = 'orders_enriched',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+) AS 
+SELECT 
+	OS.ID AS ORDER_ID,
+	OS.DISCOUNT_CODE,
+	DISCOUNT_CODES.PERCENTAGE,
+	OS.ITEM_CT,
+	OS.ORDER_SUBTOTAL, 
+	OS.ORDER_SUBTOTAL * (100 - DISCOUNT_CODES.PERCENTAGE) / 100.0 AS ORDER_TOTAL,
+	OS.ORDER_SUBTOTAL * (DISCOUNT_CODES.PERCENTAGE / 100.0) AS DISCOUNT_TOTAL
+FROM ORDER_STREAM OS 
```

Comment:
Side remark: I know it's standard SQL, but using `ORDER_STREAM OS` is IMHO hard to read and potentially error-prone. I would advocate to use `ORDER_STREAM AS OS` (if we agree to this, it might be good to audit all recipes to include the `AS` keyword for such cases). \cc @ybyzek WDYT?